### PR TITLE
Write readthedocs page contributing to ATB

### PR DESCRIPTION
**Pull request by martinghunt**

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -1,4 +1,197 @@
...
 ==============================
 
-To do
+We welcome contributions to AllTheBacteria. Please read on for instructions
+on how to contribute to the project.
+
+Background information
+----------------------
+
+Where does everything live?
```
Comment (Danderson123) [inline docs/contributing.rst:10]: Maybe better to say "Where do all the files live?" or "Where is the data located?"

File: docs/contributing.rst [hunk 2]
```diff
@@ -1,4 +1,197 @@
...
+
+* Assemblies, metadata and all analysis files are hosted at the
+  AllTheBacteria project on OSF: https://osf.io/xv7q9/.
+  ie this is where we expect users of the data to go.
+* Reproducibility files (scripts, methods descriptions) are on the
```
Comment (Danderson123) [inline docs/contributing.rst:15]: Reproducibility files does not seem immediately clear to me, maybe "Files to reproduce the analyses"

File: docs/contributing.rst [hunk 3]
```diff
@@ -1,4 +1,197 @@
...
+  AllTheBacteria GitHub repository: https://github.com/AllTheBacteria/AllTheBacteria.
+  This is where contributors should add their reproducibility files, and where
+  anyone should look for methods as opposed to the actual data.
+  The exception is large files (eg containers), which go on OSF.
```
Comment (Danderson123) [inline docs/contributing.rst:19]: containers or databases

File: docs/contributing.rst [hunk 4]
```diff
@@ -1,4 +1,197 @@
...
+To contribute, you will need:
+
+* A `GitHub <https://github.com>`_ account
+* An `OSF <https://osf.io>`_ account
+
+
+Reproducibility
+---------------
+
+Please consider reproducibility before running any analysis.
```
Comment (Danderson123) [inline docs/contributing.rst:44]: "Making your analyses reproducible"

File: docs/contributing.rst [hunk 5]
```diff
@@ -1,4 +1,197 @@
...
+knowing that it would likely break if someone else tried running it elsewhere.
+We just require that all scripts are made available, so that the process is
+clearly documented, even though it is not necessarily easy to rerun.
+For a concrete example see the
+`checkm2 analysis <https://github.com/AllTheBacteria/AllTheBacteria/tree/main/reproducibility/All-samples/checkm2>`_,
+which was run on the EBI cluster using SLURM.
+
+
+
+SOP for contributing
```
Comment (Danderson123) [inline docs/contributing.rst:63]: "Standard operating procedure (SOP)"

File: docs/contributing.rst [hunk 6]
```diff
@@ -1,4 +1,197 @@
...
+^^^^^^^^^^^^^^^^^^^^^^
+
+Run your analysis, but please bear in mind reproducibility and ease of use of
+the output files. Essentially, please be sensible, imagine you are the end-user
+of your data and think about the format that you would want it in.
+
+Please:
+
+* Take note of all commands that were run and software versions. This
+  information can be added to the methods on Github later.
```
Comment (Danderson123) [inline docs/contributing.rst:103]: Does methods mean the reproducibility files?

File: docs/contributing.rst [hunk 7]
```diff
@@ -1,4 +1,197 @@
...
+This of course may depend on your analysis, for example the
+`AMRfinderPlus results <https://osf.io/7nwrx/>`_ have an extra layer
+of versions because there is the tool version, but also the version of
+the reference database that was used.
+
+
+The intention is that reproducibility files are kept
+on GitHub, however some files will be too big. Please note:
+
+* If you used a container that is not publicly available (eg biocontainers or
```
Comment (Danderson123) [inline docs/contributing.rst:150]: maybe add a point for databases too, the AMRFinder DB is small so this is fine but the bakta DB is massive so probably not possible to upload

**Review by Danderson123 — COMMENTED** [review]
Minor wording changes and adding info about DBs

File: docs/contributing.rst (hunk 5 above)
Comment (martinghunt) [reply docs/contributing.rst:63]: going to do that in the text in the next sentence. Want to keep the heading short

**Comment by martinghunt** [comment]
Thanks, have changed all those (except SOP header)

**Comment by iqbal-lab** [comment]
This is all excellent, I'm happy with it. One thing you might add , is, where you say "provide your scripts but dont need to make it fully reproducible". You could add  "if you have bandwidth, please ask the community (via email or the MicroBio Slack channel) whether anyone is willing to help you make it reproducible." ?

**Comment by martinghunt** [comment]
@iqbal-lab have added that in. I think we're good to merge now?

**Pull request merged**